File: grpc-gen-python/example/python/src/file_system_demo_message_read.py
#! /usr/bin/python

from generated import addressbook_pb2
import sys


# Iterates though all people in the AddressBook and prints info about them.
def ListPeople(_address_book):
    for person in _address_book.people:
        print("Person ID:", person.id)
        print("  Name:", person.name)
        if person.email:
            print("  E-mail address:", person.email)

        for phone_number in person.phones:
            if phone_number.type == addressbook_pb2.Person.PhoneType.MOBILE:
                print("  Mobile phone #: ", end=' ')
            elif phone_number.type == addressbook_pb2.Person.PhoneType.HOME:
                print("  Home phone #: ", end=' ')
            elif phone_number.type == addressbook_pb2.Person.PhoneType.WORK:
                print("  Work phone #: ", end=' ')
            print(phone_number.number)


# Main procedure:  Reads the entire address book from a file and prints all
#   the information inside.
if len(sys.argv) != 2:
    print("Usage:", sys.argv[0], "ADDRESS_BOOK_FILE")
    sys.exit(-1)

address_book = addressbook_pb2.AddressBook()

# Read the existing address book.
f = open(sys.argv[1], "rb")
address_book.ParseFromString(f.read())
f.close()

ListPeople(address_book)

File: grpc-gen-dart/example/dart/bin/file_system_demo_message_read.dart
import 'dart:io';

import '../lib/generated/addressbook.pb.dart';

// Iterates though all people in the AddressBook and prints info about them.
void printAddressBook(AddressBook addressBook) {
  for (Person person in addressBook.people) {
    print('Person ID: ${person.id}');
    print('  Name: ${person.name}');
    if (person.hasEmail()) {
      print('  E-mail address:${person.email}');
    }

    for (Person_PhoneNumber phoneNumber in person.phones) {
      switch (phoneNumber.type) {
        case Person_PhoneType.MOBILE:
          print('   Mobile phone #: ');
          break;
        case Person_PhoneType.HOME:
          print('   Home phone #: ');
          break;
        case Person_PhoneType.WORK:
          print('   Work phone #: ');
          break;
        default:
          print('   Unknown phone #: ');
          break;
      }
      print(phoneNumber.number);
    }
  }
}

// Reads the entire address book from a file and prints all
// the information inside.
main(List arguments) {
  if (arguments.length != 1) {
    print('Usage: list_person ADDRESS_BOOK_FILE');
    exit(-1);
  }

  // Read the existing address book.
  File file = File(arguments.first);
  AddressBook addressBook = AddressBook.fromBuffer(file.readAsBytesSync());
  printAddressBook(addressBook);
}
